src/utils.c

```diff
@@ -173,16 +181,16 @@ void pm(int M, int N, float *A)
 
 void find_replace(const char* str, char* orig, char* rep, char* output)
 {
-    char* buffer = (char*)calloc(8192, sizeof(char));
+    char* buffer = strdup(str);
     char *p;
-
-    sprintf(buffer, "%s", str);
     if(!(p = strstr(buffer, orig))){  // Is 'orig' even in 'str'?
-        sprintf(output, "%s", str);
-        free(buffer);
+	if(str!=output) {
+	    strcpy(output,str);
+	}
         return;
     }
 
+
     *p = '\0';
```

Comment:
Will it spoil the original string `str`, since `p` is pointed to `str`-string if we don't allocate temporal `char* buffer`?

Reply:
No, p points to buf